### PR TITLE
Update tsdataset.py (issue #741)

### LINKED ISSUE #741: Make `to_dataset` method work with copy
### 🚀 Feature Request

Make TSDataset method `to_dataset` work with copy of the passed dataframe

### Motivation

Some users argues that it is not obvious that `to_dataset` method modify the dataframe inside

### Proposal

Now `to_dataset`  method should work with copy of the dataframe

### Test cases

- In `tests/test_dataset/test_dataset.py` create test `test_to_dataset_not_modify_dataframe`
- Use some fixture as an input of the `to_dataset` method and make sure it does not modify the passed dataframe

### Alternatives

_No response_

### Additional context

_No response_

### Checklist

- [ ] I discussed this issue with ETNA Team

### PR DESCRIPTION
**Pull request by mvakhmenin**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [ ] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Closing issues

closes #741

### CODE REVIEW

File: etna/datasets/tsdataset.py [hunk 1]
```diff
@@ -656,16 +656,17 @@ def to_dataset(df: pd.DataFrame) -> pd.DataFrame:
         2021-01-04           3           8
         2021-01-05           4           9
         """
-        df["timestamp"] = pd.to_datetime(df["timestamp"])
-        df["segment"] = df["segment"].astype(str)
-        feature_columns = df.columns.tolist()
+        df_copy = df.copy(deep=True)
+        df_copy["timestamp"] = pd.to_datetime(df["timestamp"])
```
Comment (martins0n) [inline etna/datasets/tsdataset.py:660]: It's better to use `df_copy` at the right  and below too

**Review by martins0n — CHANGES_REQUESTED** [review]
We should add record to CHANGELOG.md with reference to PR

**Comment by martins0n** [comment]
We should add test case as in #741  task description
image

File: etna/datasets/tsdataset.py (hunk 1 above)
Comment (mvakhmenin) [reply etna/datasets/tsdataset.py]: Done

**Review by martins0n — APPROVED** [review]
👍

**Pull request merged**